Apps/WebClient/src/ClientApp/src/components/navmenu/sidebar.vue

```diff
@@ -171,32 +174,35 @@ export default class SidebarComponent extends Vue {
         this.eventBus.$emit(EventMessageName.TimelineCreateNote);
     }
 
-    private dismissTutorial() {
-        this.logger.debug("Dismissing tutorial...");
-        if (
-            this.user.preferences.tutorialPopover != undefined &&
-            this.user.preferences.tutorialPopover.hdId != undefined
-        ) {
-            this.user.preferences.tutorialPopover.value = "false";
+    private dismissTutorial(userPreference: UserPreference) {
+        this.logger.debug(
+            `Dismissing tutorial ${userPreference.preference}...`
+        );
+        userPreference.value = "false";
+        if (userPreference.hdId != undefined) {
             this.updateUserPreference({
                 hdid: this.user.hdid,
-                userPreference: this.user.preferences.tutorialPopover,
+                userPreference: userPreference,
             });
         } else {
-            this.user.preferences.tutorialPopover = {
-                hdId: this.user.hdid,
-                preference: "tutorialPopover",
-                value: "true",
-                version: 0,
-                createdDateTime: new DateWrapper().toISO(),
-            };
+            userPreference.hdId = this.user.hdid;
             this.createUserPreference({
                 hdid: this.user.hdid,
-                userPreference: this.user.preferences.tutorialPopover,
+                userPreference: userPreference,
             });
         }
     }
 
+    private dismissTutorialForNotes() {
```

Comment:
instead of having both methods doing the same, how about passing a parameter that defines the preference?

Reply:
good idea, code updated.